xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/xwiki.postgresql.hbm.xml

```diff
@@ -454,4 +454,18 @@
             <generator class="assigned" />
         </id>
     </class>
+
+    <!-- XWikiDocumentIndexingTask custom mapping-->
+    <class name="com.xpn.xwiki.doc.tasks.XWikiDocumentIndexingTask" table="xwikidocumentindexingqueue">
+        <!-- Stores the queued tasks as well as the running tasks. -->
+        <composite-id name="id" class="com.xpn.xwiki.doc.tasks.XWikiDocumentIndexingTaskId">
+            <key-property name="docId" column="XWT_DOC_ID" type="long" />
+            <key-property name="version" column="XWT_VERSION" type="string" length="10" />
+            <key-property name="type" column="XWT_TYPE" type="string" length="100" />
+            <key-property name="instanceId" column="XWT_INSTANCE_ID" type="string" length="100" />
+        </composite-id>
```

Comment:
@tmortagne I have shortened the size of the string types, otherwise mysql is failing because the size of the index is too large (and I guess this is a problem for postgres too).

Reply:
Actually `XWT_VERSION` should be increased to 39 (2 times the length of the largest string representation of a long + 1 char for the dot).
For the other two, 100 is arbitrary but I doubt anybody will need an hint longer than 100 for instance.

Reply:
>  For the other two, 100 is arbitrary but I doubt anybody will need an hint longer than 100 for instance.

An instance ID is a UUID, right ? I think those have a max size.

Reply:
For the default implementation yes (and its 36 chars). But I we can also leave some slack for other implementations if we want to.